meson.build

```diff
@@ -39,13 +39,15 @@ cc_dict = {
     'compiler': meson.get_compiler('c'),
     'machine': host_machine,
     'config': conf,
+    'header-subdir': 'private',
```

Comment:
Would you mind renaming it into `config_h_subdir`, `buildtype_subdir` or equivalent? Header is a bit too vague.